### DISCUSSION TITLE
Cannot access ftp server deployed over docker in ec2 through load balancer from port 21

### DISCUSSION THREAD

**Discussion by ShubhTakale** [drakkan/sftpgo#1110] [Q&A]
Getting below error when trying to list the directories,
**421 Service not available, remote server has closed connection
Passive mode refused.**

**Comment by drakkan** [comment]
This has been asked many times in the past, check old issues/discussions. You need to understand how the FTP protocol works and configure your load balancer appropriately. Take also a look at the SFTPGo logs.

Check the [support policy](https://github.com/drakkan/sftpgo#support-policy) if you want less generic help. Thanks

**Reply by ShubhTakale** [reply]
Thank you for the reply!!
I checked the previous discussions but could not find the appropriate solution. Would you please provide the link of the appropriate discussion?

**Reply by matthewlenz** [reply]
He's probably referring to this https://github.com/drakkan/sftpgo/discussions/1065  which has nothing to do with AWS EC2 LB.  The EC2 LB doesn't support passing a RANGE of ports.  You have to setup the listener on all your passive ports and then create a target group for every one of those passive ports.  It'd be a complete nightmare.

**Reply by drakkan** [reply]
> He's probably referring to this #1065 which has nothing to do with AWS EC2 LB. The EC2 LB doesn't support passing a RANGE of ports. You have to setup the listener on all your passive ports and then create a target group for every one of those passive ports. It'd be a complete nightmare.

also #1068 and I remember similar questions in the past. Load balancer differs but concepts are the same

**Reply by ShubhTakale** [reply]
Thank you so much @drakkan @matthewlenz

**Comment by matthewlenz** [comment]
If you only need to allow a few active connections it might not be a big deal otherwise it's not worth you time.  Unless AWS has changed the load balancer you'd need a target group for every single passive ip in the configured range.

**Reply by ShubhTakale** [reply]
As you said we need to add a target group for every single passive ip, this means we are exposing the passive ports to internet. Isn't this a security concern?

**Reply by matthewlenz** [reply]
Read up on how FTP works.  Just use some other more modern file transfer protocol like SFTP (SSH).
